# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/ipc/ProcessPriorityManager.cpp
# Commit: 2050f2cde135
# Full Hash: 2050f2cde1353983a160cb922ca32fc8e5a5f8ea
# Author: Andrew McCreight <continuation@gmail.com>
# Date: 2021-06-11 09:42:05
# Description:
#   Bug 1715311 - Null check GetBrowserParent() in ProcessPriorityManagerImpl::ActivityChanged. r=kmag
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D117455
# ==============================================================================

diff -r 8e191c884ec1 -r 2050f2cde135 dom/ipc/ProcessPriorityManager.cpp
--- a/dom/ipc/ProcessPriorityManager.cpp	Fri Jun 11 02:46:23 2021 +0300
+++ b/dom/ipc/ProcessPriorityManager.cpp	Thu Jun 10 23:06:57 2021 +0000
@@ -467,7 +467,9 @@
     CanonicalBrowsingContext* canonical = aContext->Canonical();
     if (ContentParent* cp = canonical->GetContentParent()) {
       if (RefPtr pppm = GetParticularProcessPriorityManager(cp)) {
-        pppm->ActivityChanged(canonical->GetBrowserParent(), aIsActive);
+        if (auto* bp = canonical->GetBrowserParent()) {
+          pppm->ActivityChanged(bp, aIsActive);
+        }
       }
     }
   });
